src/Microsoft.PowerShell.Commands.Utility/commands/utility/MatchString.cs

```diff
@@ -1427,6 +1427,13 @@ protected override void BeginProcessing()
                 this.ThrowTerminatingError(errorRecord);
             }
 
+            if (this.MyInvocation.BoundParameters.ContainsKey(nameof(AllMatches)) && !this.MyInvocation.BoundParameters.ContainsKey(nameof(SimpleMatch)))
+            {
+                InvalidOperationException exception = new InvalidOperationException(MatchStringStrings.CannotSpecifyAllMatchesWithSimpleMatch);
+                ErrorRecord errorRecord = new ErrorRecord(exception, "CannotSpecifyAllMatchesWithSimpleMatch", ErrorCategory.InvalidData, null);
+                this.ThrowTerminatingError(errorRecord);
+            }
+
```

Comment:
Can we offload this check to the parameter binder by putting them in separate parameter sets? 🙂

Reply:
It already looks scary
![image](https://user-images.githubusercontent.com/22290914/69059191-dd90fd80-0a36-11ea-99e5-98afb678f9e3.png)

Reply:
Fair enough, that does look like quite a bit. We'll need to make a note to update the documentation for these parameters as well.

Reply:
Also, this condition needs to be adjusted; you have it checking for `<allmatches> && !<simplematch>` instead of `<allmatches> && <simplematch>` to throw the error. 🙂